In [3]:
#task 10.1
#This code creates a human mobility network using the data and counts the number of nodes and edges. 
import os
os.chdir(r'C:\Users\[editted]\OneDrive\Desktop\College\mining big data\MA_cbg_human_mobility_2019')

import pandas as pd
df = pd.read_csv('MA_cbg2cbg_2019_01.csv')

import networkx as nx
mobility_network = nx.Graph()
for i in range(len(df)):
    origin_cbg = df.loc[i, 'poi_cbg']
    destination_cbg = df.loc[i, 'visitor_census_tract']
    total_residents_in_origin = df.loc[i, 'raw_device_counts']
    num_visitor = df.loc[i, 'visitor_count']
    percent_visitor = num_visitor / total_residents_in_origin
    mobility_network.add_edge(origin_cbg, destination_cbg, weight=percent_visitor)
#nx.draw(mobility_network, with_labels=True)

print(nx.info(mobility_network))

Graph with 40292 nodes and 742299 edges


C:\Users\amyli\AppData\Local\Temp\ipykernel_10020\2936956621.py:20: DeprecationWarning: info is deprecated and will be removed in version 3.0.

  print(nx.info(mobility_network))


TASK 10.1
There 40292 nodes and 742299 edges in my human mobility network. 

In [ ]:
#task 10.2
#This code creates a new dataframe with three columns: census block group id, weighted node degree, and unweighted node degree. 
weighted_list = []
unweighted_list = []
for element in df['poi_cbg']:
    unweighted_list.append(mobility_network.degree(element))
    weighted_list.append(mobility_network.degree(element, weight='weight'))

new_df = pd.DataFrame(df, columns=['census_block_group_id','weighted_node_degree','unweighted_node_degree'])
group_id = df['poi_cbg']
new_df['census_block_group_id'] = group_id
new_df['weighted_node_degree'] = weighted_list
new_df['unweighted_node_degree'] = unweighted_list
new_df

In [ ]:
#task 10.3
#This code creates a histogram of the weighted node degrees in the new dataframe
import seaborn as sns
#sns.barplot(x = 'census_block_group_id', y = 'weighted_node_degree', data = new_df)
plot = sns.histplot(data = new_df, x = new_df['weighted_node_degree'].astype(str))
plot

TASK 10.3
The plot does not show the degree distribution follow a power law, because there is no decreasing pattern in the histogram. 